synapse/storage/databases/main/devices.py

```diff
@@ -1614,20 +1618,21 @@ async def check_too_many_devices_for_user(self, user_id: str) -> List[str]:
 
         # Now fetch the devices to delete.
         sql = """
-            SELECT DISTINCT device_id FROM devices
+            SELECT device_id FROM devices
```

Comment:
Why no longer DISTINCT?

Reply:
oh, presumably (user_id, device_id) is already unique

Reply:
It was pointless mostly, as there are unique indexes on `user_id / device_id`